packages/client/machines/appMachine.ts

```diff
@@ -291,119 +291,198 @@ export function createAppMachine({
+                            initial: 'pollingMyProfileInformation',
 
-                                onDone: {
-                                    target: 'revalidatedUserInformation',
+                            states: {
+                                pollingMyProfileInformation: {
+                                    invoke: {
+                                        src: 'fetchUser',
 
-                                    actions: assign({
-                                        myProfileInformation: (_context, e) => {
-                                            const event =
-                                                e as DoneInvokeEvent<GetMyProfileInformationResponseBody>;
+                                        onDone: [
+                                            {
+                                                cond: (_context, e) => {
+                                                    const event =
+                                                        e as DoneInvokeEvent<GetMyProfileInformationResponseBody>;
+                                                    return (
+                                                        event.data
+                                                            .hasConfirmedEmail ===
+                                                        true
+                                                    );
+                                                },
+                                                actions: assign({
+                                                    myProfileInformation: (
+                                                        _context,
+                                                        e,
+                                                    ) => {
+                                                        const event =
+                                                            e as DoneInvokeEvent<GetMyProfileInformationResponseBody>;
+
+                                                        return event.data;
+                                                    },
+                                                }),
+                                                target: 'userEmailIsVerified',
+                                            },
+                                            {
+                                                target: 'debouncing',
+                                            },
+                                        ],
 
-                                            return event.data;
+                                        onError: {
+                                            target: 'debouncing',
                                         },
-                                    }),
+                                    },
+                                },
+
+                                debouncing: {
+                                    after: {
+                                        500: {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏I think 500ms is waaaaay too short. 5s would be better. I think it's better to let users wait some seconds than making two requests per second in background, for an undetermined amount of time. The application could be made slow for users with a bad internet connection.